engine/src/main/java/org/terasology/engine/world/chunks/remoteChunkProvider/ReceivedChunkInitialProvider.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.world.chunks.remoteChunkProvider;
+
+import org.joml.Vector3ic;
+import org.terasology.engine.world.chunks.Chunk;
+import org.terasology.engine.world.chunks.pipeline.InitialChunkProvider;
+
+import java.util.Comparator;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.PriorityBlockingQueue;
+
+public class ReceivedChunkInitialProvider implements InitialChunkProvider {
+    private BlockingQueue<Chunk> chunkQueue;
+
+    public ReceivedChunkInitialProvider(Comparator<Chunk> comparator) {
+        chunkQueue = new PriorityBlockingQueue<>(800, comparator);
```

Comment:
I guess the 800 is any arbitrary number? Or is there more math behind it?

Reply:
That's the initial capacity that was used for the priority queue in the previous ChunkProcessingPipeline; it probably doesn't actually need to be that high here.